packages/mobile/src/exchange/ExchangeReview.tsx

```diff
@@ -1,133 +1,229 @@
+    let amount = this.state.inputAmount
+    if (this.state.inputToken !== token) {
+      const conversionRate = getRateForMakerToken(
+        this.props.exchangeRatePair,
+        this.state.makerToken,
+        this.state.inputToken
+      )
+      amount = getTakerAmount(amount, conversionRate)
+    }
+    return amount
   }
 
   render() {
-    const { exchangeRatePair, fee, t, appConnected, dollarBalance, goldBalance } = this.props
-    const { makerToken, makerAmount } = this.getConfirmationInput()
-    const rate = getRateForMakerToken(exchangeRatePair, makerToken)
-    const takerAmount = getTakerAmount(makerAmount, rate)
-    const newGoldBalance = getNewGoldBalance(goldBalance, makerToken, makerAmount, takerAmount)
-    const newDollarBalance = getNewDollarBalance(
-      dollarBalance,
-      makerToken,
-      makerAmount,
-      takerAmount
+    const { exchangeRatePair, fee, t, appConnected } = this.props
+
+    const exchangeRate = getRateForMakerToken(
+      exchangeRatePair,
+      this.state.makerToken,
+      CURRENCY_ENUM.DOLLAR
     )
+    const dollarAmount = this.getInputAmountInToken(CURRENCY_ENUM.DOLLAR)
+
+    const buttonText =
+      this.state.makerToken === CURRENCY_ENUM.DOLLAR
+        ? t(`${Namespaces.exchangeFlow9}:buy`)
+        : t(`${Namespaces.exchangeFlow9}:sell`)
 
     return (
       <SafeAreaView style={styles.container}>
-        <DisconnectBanner />
-        <ReviewFrame
-          HeaderComponent={this.renderHeader}
-          confirmButton={{
-            action: this.onPressConfirm,
-            text: t('exchange'),
-            disabled: !appConnected || rate.isZero(),
-          }}
-          modifyButton={{ action: this.onPressEdit, text: t('edit') }}
-        >
-          <ExchangeConfirmationCard
-            makerToken={makerToken}
-            newDollarBalance={newDollarBalance}
-            newGoldBalance={newGoldBalance}
-            makerAmount={makerAmount}
-            takerAmount={takerAmount}
-            exchangeRate={rate}
-            fee={fee}
+        <View style={styles.paddedContainer}>
+          <DisconnectBanner />
+          <ScrollView>
+            <View style={styles.column}>
+              <View style={[styles.rowContainer, styles.amountRow]}>
+                <Text style={[fontStyles.body, styles.exchangeBodyText]}>
+                  {t('amount') + ` (${this.state.inputTokenCode})`}
+                </Text>
+                <Text style={[fontStyles.body, styles.currencyAmountText]}>
+                  {getMoneyDisplayValue(this.state.inputAmount, this.state.inputToken, true)}
+                </Text>
+              </View>
+              <View style={styles.line} />
+              <View style={[styles.rowContainer, styles.feeRowContainer]}>
+                <Text style={[fontStyles.body, styles.exchangeBodyText]}>
+                  {t('subtotal') +
+                    ' @ ' +
+                    getMoneyDisplayValue(exchangeRate, CURRENCY_ENUM.DOLLAR, true)}
```

Comment:
Same comment about using interpolation instead of concatenation which can't be translated correctly.